src/generator.rs

```diff
@@ -90,6 +91,31 @@ pub fn run_generator(devices: &[Device], output_directory: &Path, fake_mode: boo
                 < max_device
             {}
         }
+
+        let compressions: Vec<&String> = devices
+            .iter()
+            .filter(|device| !device.compression_algorithm.is_none())
+            .map(|device| device.compression_algorithm.as_ref().unwrap())
```

Comment:
This is just `.flat_map(|d| d.compression_algorithm.as_ref())`.

Reply:
Duh.